src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -75,61 +82,76 @@ public GlobbingUrlBuilder(IFileProvider fileProvider, IMemoryCache cache, PathSt
         /// <returns>The list of URLs</returns>
         public virtual ICollection<string> BuildUrlList(string staticUrl, string includePattern, string excludePattern)
         {
-            var urls = new HashSet<string>(StringComparer.Ordinal);
 
-            // Add the statically declared url if present
+            var urls = new List<string>();
             if (staticUrl != null)
             {
                 urls.Add(staticUrl);
             }
 
             // Add urls that match the globbing patterns specified
-            var matchedUrls = ExpandGlobbedUrl(includePattern, excludePattern);
-            urls.UnionWith(matchedUrls);
```

Comment:
Ran this by @dougbu, it feels like the code wanted distinct values in an order preserving collection. HashSet happened to preserve the order but this formalizes it.